Fix the Buildkite agent module

jamesottaway opened the pull request:
I tried the following config in my `/etc/macos/configuration.nix`:

```nix
{
  services.buildkite-agent = {
    enable = true;
    tokenPath = ./token;
    openssh = {
      publicKeyPath = "/dev/null";
      privateKeyPath = "/dev/null";
    };
  };

  users.knownUsers = [ "buildkite-agent" ];
  users.users.buildkite-agent = {
    createHome = true;
    uid = 1009;
  };
}
```

When I ran `darwin-rebuild switch` I found the following errors in `syslog`:

```
❯ syslog | rg buildkite                                                                                                             ~
NOTE:  Most system logs have moved to a new logging system.  See log(1) for more information.
Oct 10 14:16:09 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[145]) <Warning>: Could not find user/group associated with service: 0: Undefined error: 0 buildkite-agent/buildkite-agent
Oct 10 14:16:09 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[145]) <Notice>: Service setup event to handle failure and will not launch until it fires.
Oct 10 14:16:09 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[145]) <Warning>: Service exited with abnormal code: 78
Oct 10 15:31:10 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[3247]) <Warning>: Could not find user/group associated with service: 0: Undefined error: 0 buildkite-agent/buildkite-agent
Oct 10 15:31:10 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[3247]) <Notice>: Service setup event to handle failure and will not launch until it fires.
Oct 10 15:31:10 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[3247]) <Warning>: Service exited with abnormal code: 78
Oct 11 12:05:14 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent) <Error>: The NetworkState key is no longer respected. Please remove it.
Oct 11 12:05:14 b12y-MBP com.apple.xpc.launchd[1] (com.apple.xpc.launchd.user.domain.501.100022.Aqua) <Error>: org.nixos.buildkite-agent (lint): UserName is not supported for non-System services.
Oct 11 12:05:14 b12y-MBP com.apple.xpc.launchd[1] (com.apple.xpc.launchd.user.domain.501.100022.Aqua) <Error>: org.nixos.buildkite-agent (lint): GroupName is not supported for non-System services.
Oct 11 12:05:14 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[28816]) <Error>: Service could not initialize: 19H2: xpcproxy + 14521 [839][C9A1BADB-35DE-3593-881A-1C75B5108198]: 0xd
Oct 11 12:05:14 b12y-MBP com.apple.xpc.launchd[1] (org.nixos.buildkite-agent[28816]) <Warning>: Service exited with abnormal code: 78
```

By removing these 3 lines I was able to get it working.

LnL7 commented:
That seems undesirable as this will make the entire service run as root. Your example config only created the user and not the group, perhaps that was the problem?

jamesottaway commented:
Ah, I was looking at the `UserName is not supported for non-System services` line but I missed the `Could not find user/group associated with service: 0: Undefined error: 0 buildkite-agent/buildkite-agent` warning above that.

I reverted my change to this file, updated my config to create the group, ran `darwin-rebuild switch` and (after tweaking some file/folder permissions, and manually starting the daemon) it worked.

Here's the config I ended up with:

```
{
  services.buildkite-agent = {
    enable = true;
    tokenPath = ./token;
    extraConfig = ''
      no-pty=true
    '';
    openssh = {
      publicKeyPath = /dev/null;
      privateKeyPath = /dev/null;
    };
  };

  users.knownUsers = [ "buildkite-agent" ];
  users.users.buildkite-agent.createHome = true;
  users.users.buildkite-agent.uid = 1009;

  users.knownGroups = [ "buildkite-agent" ];
  users.groups.buildkite-agent.gid = 1009;
}
```

Just to confirm, since I couldn't find any docs stating it: is it up to everyone individually to manage their known users/groups? Why can't the module mark it as known?
